Adding edited emails.

Anish-Agnihotri opened the pull request:
##### Description

This PR introduces one email template (update_notification). It is fully completed, and 100% mobile responsive. In a future PR, I will be adding its routing into Django.

*new:* In addition, this also fixes common issues across all email files, and resolves any pre-existing merge errors. 

##### Checklist

- [x] Linter status: 100% pass (failing checks are because of other files.). 
- [x] Changes don't break existing behavior
- [x] Commit message follows [commit guidelines](https://github.com/gitcoinco/web/blob/master/docs/CONTRIBUTING.md#step-4-commit)
- [x] Checklist for #2289 has been met

##### Affected core subsystem(s)

No core systems have been affected until the email is implemented into the respective `emails.py` and `urls.py` files. I will be completing this integration as well after this PR by using dummy variables.

##### Testing
This change has been tested across two operating systems using multiple integration tests. In order to test for yourself, you have two options: 

1. Since the email has not been implemented into `urls.py`, you can replace an existing email (I personally used `new_bounty.html`) with the content of the pushed email to see data.
2. You can setup the routing yourself and see the magic happen. 

##### Refers/Fixes

Resolves the following issues: 

Update Notification, #2289, #2088

mbeacom commented on app/retail/templates/emails/grants/support_cancellation.html:
```diff
@@ -60,126 +64,163 @@
+  </a>
+  <!-- width="50" height="50"Input project image here -->
+  <br>
+  <a class="grantInfoName" href="https://github.com/{{ github_username }}">{{ grant_name }}</a>
```
Shouldn't we link to the grant details page here?

mbeacom commented on app/retail/templates/emails/grants/support_cancellation.html:
```diff
@@ -1,56 +1,60 @@
+{% extends '../emails/template.html' %} 
+{% comment %} 
```
Trailing whitespace.

mbeacom commented on app/retail/templates/emails/grants/subscription_terminated.txt:
```diff
@@ -1,21 +1,21 @@
 {% load i18n humanize %}
 
-{% trans "Grant" %} {{grant_name}} {% trans "has been cancelled" %}
+{% trans "Grant" %} {{ grant_name }} {% trans "has been cancelled" %}
```
All of these variables need updated to take the subscription object and use it's associated attributes.

mbeacom commented on app/retail/templates/emails/grants/subscription_terminated.html:
```diff
@@ -60,93 +64,117 @@
-<a class="updatesButton" href="https://etherscan.com/{{etherscan_link}}">{%trans "View Transaction" %}</a> <!-- Input link to view transaction button here --> 
-<br>
-
+<a class="updatesButton" href="https://etherscan.com/{{ etherscan_link }}">{%trans "View Transaction" %}</a>
```
All of the `trans` blocks need fixed for whitespace...

mbeacom commented on app/retail/templates/emails/grants/subscription_terminated.html:
```diff
@@ -60,93 +64,117 @@
+  <p>{{ grant_description }}</p>
+  <!-- Input project description here -->
+  <div class="progress">
+    <h3 style="text-transform: none;">{{ funding_status }}</h3>
```
All of the template vars throughout this file need to reference the `Grant` object and it's associated attributes.

mbeacom commented on app/retail/templates/emails/grants/subscription_terminated.html:
```diff
@@ -60,93 +64,117 @@
+  </a>
+  <!-- width="50" height="50"Input project image here -->
+  <br>
+  <a class="grantInfoName" href="https://github.com/{{ github_username }}">{{ github_username }}</a>
```
This should be linking back to the grant page vs the user's github page.

mbeacom commented on app/retail/templates/emails/grants/subscription_terminated.html:
```diff
@@ -60,93 +64,117 @@
-.funding > p {
+  }
+
+  .funding>p {
```
?

mbeacom commented on app/retail/templates/emails/grants/subscription_terminated.html:
```diff
@@ -1,56 +1,60 @@
 {% endcomment %}
-{% load i18n humanize %}
-{% load i18n static %}
+{% load i18n humanize static %} 
```
No need for humanize in here.

mbeacom commented on app/retail/templates/emails/grants/subscription_terminated.html:
```diff
@@ -1,56 +1,60 @@
+{% extends '../emails/template.html' %} 
+{% comment %}   
```
whitespace

mbeacom commented on app/retail/templates/emails/grants/new_supporter.txt:
```diff
@@ -2,10 +2,9 @@
+{{ variable_amount_contribution }}
 
-{%trans "You can see a transaction history and withdraw funds" %} {{transaction_history}} {%trans "here." %}
+{%trans "You can see transaction history" %} {{ transaction_history }} {%trans "here." %}
```
trans whitespace needs fixed throughout

mbeacom commented on app/retail/templates/emails/grants/new_supporter.txt:
```diff
@@ -2,10 +2,9 @@
 {% trans "You have a new supporter!" %}
 
-{{github_username}}
+{{ github_username }}
```
All of these vars should reference the input `Grant` object...

mbeacom commented on app/retail/templates/emails/grants/new_supporter.html:
```diff
@@ -1,94 +1,90 @@
-<br>
+<!-- Input link to transaction history here -->
+<p>{%trans "You can see transaction history" %}
+  <a href="https://gitcoin.co/{{ transaction_history }}">{%trans "here." %}</a>
```
This needs to use `{% url %}` references vs hardcoded paths.  - Throughout all of these files.

mbeacom commented on app/retail/templates/emails/grants/new_supporter.html:
```diff
@@ -1,94 +1,90 @@
-{% extends 'emails/template.html' %}
```
Unnecessary trailing whitespace, vars need to use the input `subscription` and/or `grant` objects.  None of these variables will exist.

mbeacom commented on app/retail/templates/emails/grants/new_supporter.html:
```diff
@@ -1,94 +1,90 @@
+  along with this program. If not, see <http://www.gnu.org/licenses/>.
+  
+{% endcomment %} 
+{% load i18n humanize static %} 
```
No need for humanize here

mbeacom commented on app/retail/templates/emails/grants/grant_cancellation.html:
```diff
@@ -1,46 +1,44 @@
-{% extends 'emails/template.html' %}
```
No reason for the trailing whitespace additions here.
All vars need updated to use the input grant object, none of these variables will exist.
Follow django proper whitespace for template variables and functions.
URLs need updated to use the `url` reverse.
We need all of these templates hooked up to the respective email func and passing the grant and/or subscription objects into the email methods on event.

Anish-Agnihotri commented:
Thanks for review @mbeacom. Always appreciated. I'll refactor and resubmit within 24hr.

mbeacom commented:
@Anish-Agnihotri Is there anything I can do to help you with this?  Are you still interested in moving this forward?

Anish-Agnihotri commented:
@mbeacom For sure! Been done with a fever so have been unable to refactor. Thanks a bunch for the review, I'll be pushing out a new commit within the next day.

Anish-Agnihotri commented:
Just fixed the other email. This PR should be updated within the next day, if not the next few hours. Just figuring out some small things in `urls.py` and making sure I move over the objects to their appropriate `grant` and `subscription` objects.

Other than that, just basic aesthetic changes.

Anish-Agnihotri commented:
Status check on commit (updating as I go): 

Grant Cancellation Email: 👍 
1. Removed trailing white-spaces.
2. Updated to use grants object `{{ grants.title }}`.
3. Updated Django tag whitespacing. 
4. Removed unnecessary humanize load since no numbers were being parsed. 

Grants Email: 
New Supporter Email: 
Subscription Terminated Email:
Support Cancellation Email: 
Thank You For Supporting Email:
Update Notification Email:

Anish-Agnihotri commented:
Ready for re-review @mbeacom @captnseagraves.

Thanks!
